presto-main/src/main/java/com/facebook/presto/dynamicCatalog/DynamicCatalogController.java

```diff
@@ -0,0 +1,179 @@
+import java.util.LinkedHashSet;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+
+import static com.google.common.base.Strings.nullToEmpty;
+import static io.airlift.discovery.client.ServiceAnnouncement.serviceAnnouncement;
+
+@Path("/v1/catalog")
+public class DynamicCatalogController
+{
+    private final CatalogManager catalogManager;
+    private final ConnectorManager connectorManager;
+    private final Announcer announcer;
+    private final InternalNodeManager internalNodeManager;
+    private final File catalogConfigurationDir;
+    private static final Logger log = Logger.get(DynamicCatalogController.class);
+    private final ResponseParser responseParser;
+
+    private DynamicCatalogController(CatalogManager catalogManager, ConnectorManager connectorManager, Announcer announcer, InternalNodeManager internalNodeManager, File catalogConfigurationDir, ResponseParser responseParser)
+    {
+        this.catalogManager = catalogManager;
+        this.connectorManager = connectorManager;
+        this.announcer = announcer;
+        this.internalNodeManager = internalNodeManager;
+        this.catalogConfigurationDir = catalogConfigurationDir;
+        this.responseParser = responseParser;
+    }
+
+    @Inject
+    public DynamicCatalogController(CatalogManager catalogManager, ConnectorManager connectorManager, Announcer announcer, InternalNodeManager internalNodeManager, StaticCatalogStoreConfig config, ResponseParser responseParser)
+    {
+        this(catalogManager, connectorManager, announcer, internalNodeManager, config.getCatalogConfigurationDir(), responseParser);
+    }
+
+    @Path("add")
+    @POST
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response addCatalog(CatalogVo catalogVo)
+    {
+        try {
+            log.info("addCatalog : input values are " + catalogVo);
+            ConnectorId connectorId = connectorManager.createConnection(catalogVo.getCatalogName(), catalogVo.getConnectorName(), catalogVo.getProperties());
+            updateConnectorIdAnnouncement(announcer, connectorId, internalNodeManager);
+            log.info("addCatalog() : catalogConfigurationDir: " + catalogConfigurationDir.getAbsolutePath());
+            writeToFile(catalogVo);
+            log.info("addCatalog() : Successfully added catalog " + catalogVo.getCatalogName());
+            return successResponse(responseParser.build("Successfully added catalog: " + catalogVo.getCatalogName(), 200));
+        }
+        catch (Exception ex) {
+            log.error("addCatalog() : Error adding catalog " + ex.getMessage());
+            return failedResponse(responseParser.build("Error adding Catalog: " + ex.getMessage(), 500));
+        }
+    }
+
+    @Path("delete")
+    @DELETE
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response deleteCatalog(@QueryParam("catalogName") String catalogName)
```

Comment:
It looks like this will try to delete whatever catalog is in `StaticCatalogStoreConfig`.  I think it would be operationally better to let whatever is specified as a static resource to be immutable, but still allow one to delete or add on the fly.  I don't think it should delete the files that are statically shipped with the Presto deployment.  Worst case scenario, you don't specify any static catalogs and you need to add them via the REST endpoint.  But that seems better than accidentally deleting an important catalog config and effectively not having a way to undo that.

Reply:
Thanks for the comment,
Not sure I followed your comment still I tried to answer.
Delete catalog endpoint  will call ```connectorManager.dropConnection(catalogName);``` which will eventually call ```removeCatalog``` method of ```catalogManager```  which deletes the catalog from concurrentMap.
The endpoint will also delete the file associated with this catalog from etc/catalog which was created when we registered the catalog.

Reply:
Are the catalogs that this endpoint allows one to delete guaranteed to be registered through the same endpoint?  If not, then I am questioning if it's really desirable to allow one to delete catalog configs which were statically loaded from `etc/catalog`.  It seems like it's better, at the very least, to load and unload them in an isolated directory, and whatever is loaded on startup in `etc/catalog` should be immutable.

Reply:
The endpoint deletes catalogs irrespective of added using the endpoint or already existing , because any new catalog will be added to etc/catalog and will be removed from same directory.

When presto is being install first time, it doesn't have any etc/catalog folder ,so there are no catalogs provided as a part of installation , it is up to the user to add whatever catalog they want to use, hence there is nothing to be immutable.
If the question is deletion existing catalog during development,there is already the etc/catalog folder checked in presto main which will sync back the etc/catalog.